pkg/cidr/cidr.go

```diff
@@ -135,3 +135,55 @@ func MustParseCIDR(str string) *CIDR {
 	}
 	return c
 }
+
+// Version returns the IP version for an IP, or 0 if the IP is not valid.
+func IPVersion(i *net.IP) int {
+	if i.To4() != nil {
+		return 4
+	} else if len(*i) == net.IPv6len {
+		return 6
+	}
+	return 0
+}
+
+// matchCIDRs matchs an IP address against a list of cidrs.
+// If the list is empty, it always matches.
+func matchCIDRs(ip net.IP, cidrs []*net.IPNet) bool {
+	if len(cidrs) == 0 {
+		return true
+	}
+	for _, cidr := range cidrs {
+		if cidr.Contains(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+func DetectCIDRByCIDR(cidr *CIDR, version int) (*net.Interface, net.IP, *net.IPNet, error) {
+	netIfaces, err := net.Interfaces()
```

Comment:
FYI if we do not remove this code while addressing the other feedback on this PR, then we will want to consider replacing `net.Interfaces()` here with `netlink.LinkList()` for the same reason as https://github.com/cilium/cilium/pull/15260.

Reply:
Updated accordingly. Thanks